src/output.rs

```diff
@@ -159,156 +130,35 @@ impl State {
-
-            #[cfg(feature = "stm32f3")]
-            type GpioX = hal::gpio::$GpioX;
-            #[cfg(feature = "stm32f3")]
-            type Afr = <Self as hal::gpio::marker::IntoAf<{ $CHY_AF }>>::AFR;
-
-            // Pin<Gpio, Index, Alternate<PushPull, AF>>
-            fn connect_to_hrtim(
-                self,
-                #[cfg(feature = "stm32f3")]
-                moder: &mut <Self::GpioX as hal::gpio::marker::GpioStatic>::MODER,
-                #[cfg(feature = "stm32f3")]
-                otyper: &mut <Self::GpioX as hal::gpio::marker::GpioStatic>::OTYPER,
-                #[cfg(feature = "stm32f3")] afr: &mut Self::Afr,
-            ) {
-                #[allow(non_snake_case, unused_variables)]
-                let $GpioX = ();
-
-                #[cfg(feature = "stm32f3")]
-                let _: $CHY<hal::gpio::Alternate<hal::gpio::PushPull, $CHY_AF>> =
-                    self.into_af_push_pull(moder, otyper, afr);
-
-                #[cfg(any(feature = "stm32g4", feature = "stm32h7"))]
-                let _: $CHY<hal::gpio::Alternate<{ $CHY_AF }>> = self.into_alternate();
-            }
-        }
-    };
-}
-
-// $GpioX is only used for f3x4
-macro_rules! pins {
-    ($($TIMX:ty: CH1: $CH1:ident<$CH1_AF:literal>, CH2: $CH2:ident<$CH2_AF:literal>, $GpioX:ident)+) => {$(
-        pins_helper!($TIMX, HrOut1, $CH1<$CH1_AF>, $GpioX);
-        pins_helper!($TIMX, HrOut2, $CH2<$CH2_AF>, $GpioX);
-    )+};
-}
-
-#[cfg(any(feature = "stm32g4", feature = "stm32f3"))]
-pins! {
-    HRTIM_TIMA: CH1: PA8<13>,  CH2: PA9<13>,  Gpioa
-    HRTIM_TIMB: CH1: PA10<13>, CH2: PA11<13>, Gpioa
-    HRTIM_TIMC: CH1: PB12<13>, CH2: PB13<13>, Gpiob
-    HRTIM_TIMD: CH1: PB14<13>, CH2: PB15<13>, Gpiob
-    HRTIM_TIME: CH1: PC8<3>,   CH2: PC9<3>,   Gpioc
-}
-
-#[cfg(feature = "stm32g4")]
-pins! {
-    HRTIM_TIMF: CH1: PC6<13>, CH2: PC7<13>, Gpioc
-}
-
-// TODO: H7 does not start in input mode, it starts in Analog
-#[cfg(feature = "stm32h7")] // RM0433
-pins! {
-    HRTIM_TIMA: CH1: PC6<1>,  CH2: PC7<1>,  Gpioc
-    HRTIM_TIMB: CH1: PC8<1>, CH2: PA8<2>, GpioC // This type is not used for in this config so it's ok
-    HRTIM_TIMC: CH1: PA9<2>, CH2: PA10<2>, GpioA
-    HRTIM_TIMD: CH1: PA11<2>, CH2: PA12<2>, GpioA
-    HRTIM_TIME: CH1: PG6<2>,  CH2: PG7<2>,   GpioG
-}
-
-impl<T> sealed::Sealed<T> for () {}
-impl<T> ToHrOut<T> for () {
+unsafe impl<T> ToHrOut<T> for () {
     type Out<PSCL> = ();
-    #[cfg(feature = "stm32f3")]
-    type GpioX = hal::gpio::Gpioa;
-    #[cfg(feature = "stm32f3")]
-    type Afr = ();
-
-    fn connect_to_hrtim(
-        self,
-        #[cfg(feature = "stm32f3")]
-        _moder: &mut <Self::GpioX as hal::gpio::marker::GpioStatic>::MODER,
-        #[cfg(feature = "stm32f3")]
-        _otyper: &mut <Self::GpioX as hal::gpio::marker::GpioStatic>::OTYPER,
-        #[cfg(feature = "stm32f3")] _afr: &mut Self::Afr,
-    ) {
-    }
+    type Arg = ();
+
+    fn connect_to_hrtim(self, _arg: &mut ()) {}
```

Comment:
We can probably completely remove `Arg` and `connect_to_hrtim` and let that be the hals problem